packages/backend/src/open_payments/quote/service.ts

```diff
@@ -61,36 +61,45 @@ async function getQuote(
   return Quote.query(deps.knex).get(options).withGraphFetched('asset')
 }
 
-export interface CreateQuoteOptions {
+interface QuoteOptionsBase {
   paymentPointerId: string
-  sendAmount?: Amount
-  receiveAmount?: Amount
   receiver: string
   clientId?: string
 }
 
+interface QuoteOptionsWithSendAmount extends QuoteOptionsBase {
+  sendAmount: Amount
+}
+
+interface QuoteOptionsWithReceiveAmount extends QuoteOptionsBase {
+  receiveAmount: Amount
+}
+
+export type CreateQuoteOptions =
+  | QuoteOptionsBase
+  | QuoteOptionsWithSendAmount
+  | QuoteOptionsWithReceiveAmount
+
 async function createQuote(
   deps: ServiceDependencies,
   options: CreateQuoteOptions
 ): Promise<Quote | QuoteError> {
-  if (options.sendAmount && options.receiveAmount) {
-    return QuoteError.InvalidAmount
-  }
   const paymentPointer = await deps.paymentPointerService.get(
     options.paymentPointerId
   )
   if (!paymentPointer) {
     return QuoteError.UnknownPaymentPointer
   }
-  if (options.sendAmount) {
+  if ('sendAmount' in options) {
```

Comment:
I don't think these are necessary with the suggested type edit
```suggestion
  if (options.sendAmount) {
```